### PR TITLE
Add customBoxShape

### PR DESCRIPTION
**Pull request by Dominic-go**
## Ticket
- [Custom BoxShape](https://www.notion.so/CustomBoxShape-ec39530cf2584949869249f3bbda6166?pvs=4)

## Task
- [x] Custom BoxShape

### CODE REVIEW

File: gallery_app/lib/src/widgets/box_shape.dart [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+                  (index) => Expanded(
+                        child: Container(
+                          color: index % 2 == 0
+                              ? AppColors.transparent
+                              : AppColors.gray_400,
+                          height: 2,
+                          width: 1,
+                        ),
+                      )),
+            ),
```
Comment (drake-ho-goldenowl) [inline gallery_app/lib/src/widgets/box_shape.dart]: Please use the Custom Widget to make this

File: gallery_app/lib/src/widgets/box_shape.dart [hunk 2]
```diff
@@ -17,9 +17,10 @@ class _XBoxShapeState extends State<XBoxShape> {
       child: Container(
         width: 400,
         height: 100,
-        decoration: BoxDecoration(
-            color: AppColors.slate_50,
-            border: Border.all(color: AppColors.gray_400, width: 1)),
+        decoration: const BoxDecoration(
+          color: AppColors.slate_50,
+          // border: Border.all(color: AppColors.gray_400, width: 1)
```
Comment (drake-ho-goldenowl) [inline gallery_app/lib/src/widgets/box_shape.dart]: Remove useless code

Comment (Dominic-go) [reply gallery_app/lib/src/widgets/box_shape.dart]: Ummm. Conflict file is gallery_app/lib/src/features/button/view/button_view.dart. In this feature, I only used it to display the UI of XBoxShape... At dev branch it's the button feature

File: gallery_app/lib/src/features/button/view/button_view.dart [hunk 3]
```diff
@@ -1,10 +1,23 @@
...
+import 'package:gallery_app/src/widgets/box_shape.dart';
 
 class ButtonView extends StatelessWidget {
   const ButtonView({super.key});
 
   @override
   Widget build(BuildContext context) {
-    return const Text('ButtonView');
+    return Scaffold(
+      appBar: const XAppBar(title: 'abc'),
```
Comment (drake-ho-goldenowl) [inline gallery_app/lib/src/features/button/view/button_view.dart]: Don't set name has no meaning

**Review by drake-ho-goldenowl — APPROVED** [review]

**Pull request merged**